app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/LocalFileTransferFragment.kt

```diff
@@ -83,43 +91,45 @@ class LocalFileTransferActivity : BaseActivity(),
   private var fileListAdapter: FileListAdapter? = null
   private var wifiPeerListAdapter: WifiPeerListAdapter? = null
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_local_file_transfer)
-    /*
-     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
-     * - On the sender device, this activity is started from the app chooser post selection
-     * of files to share in the Library
-     * - On the receiver device, the activity is started directly from within the 'Get Content'
-     * activity, without any file Uris
-     * */
-    val filesIntent = intent
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View = inflater.inflate(R.layout.fragment_local_file_transfer, container, false)
+
+  override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+    inflater.inflate(R.menu.wifi_file_share_items, menu)
+    super.onCreateOptionsMenu(menu, inflater)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    setHasOptionsMenu(true)
+    val activity = requireActivity() as CoreMainActivity
     val fileUriArrayList: ArrayList<Uri>? =
-      filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM)
+      if (arguments == null) arrayListOf()
+      else LocalFileTransferFragmentArgs.fromBundle(requireArguments())
+        .uris.toCollection(ArrayList())
```

Comment:
toCollection should be unnecessary and the `if` should be as well, null safe and elvis it